phpBB/phpbb/storage/controller/attachment.php

```diff
@@ -0,0 +1,503 @@
+		$this->auth->acl($this->user->data);
+		$this->user->setup('viewtopic');
+
+		if (!$this->config['allow_attachments'] && !$this->config['allow_pm_attach'])
+		{
+			throw new http_exception(404, 'ATTACHMENT_FUNCTIONALITY_DISABLED');
+		}
+
+		if (!$attach_id)
+		{
+			throw new http_exception(404, 'NO_ATTACHMENT_SELECTED');
+		}
+
+		$sql = 'SELECT attach_id, post_msg_id, topic_id, in_message, poster_id, is_orphan, physical_filename, real_filename, extension, mimetype, filesize, filetime
+			FROM ' . ATTACHMENTS_TABLE . "
+			WHERE attach_id = $attach_id";
+		$result = $this->db->sql_query($sql);
+		$attachment = $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		if (!$attachment)
+		{
+			throw new http_exception(404, 'ERROR_NO_ATTACHMENT');
+		}
+		else if (!$this->download_allowed())
+		{
+			throw new http_exception(403, 'LINKAGE_FORBIDDEN');
+		}
+
+		$attachment['physical_filename'] = utf8_basename($attachment['physical_filename']);
+
+		if (!$attachment['in_message'] && !$this->config['allow_attachments'] || $attachment['in_message'] && !$this->config['allow_pm_attach'])
+		{
+			throw new http_exception(404, 'ATTACHMENT_FUNCTIONALITY_DISABLED');
+		}
+
+		if ($attachment['is_orphan'])
+		{
+			// We allow admins having attachment permissions to see orphan attachments...
+			$own_attachment = ($this->auth->acl_get('a_attach') || $attachment['poster_id'] == $this->user->data['user_id']) ? true : false;
+
+			if (!$own_attachment || ($attachment['in_message'] && !$this->auth->acl_get('u_pm_download')) || (!$attachment['in_message'] && !$this->auth->acl_get('u_download')))
+			{
+				throw new http_exception(404, 'ERROR_NO_ATTACHMENT');
+			}
+
+			// Obtain all extensions...
+			$extensions = $this->cache->obtain_attach_extensions(true);
+		}
+		else
+		{
+			if (!$attachment['in_message'])
+			{
+				$this->phpbb_download_handle_forum_auth($attachment['topic_id']);
+
+				$sql = 'SELECT forum_id, post_visibility
+					FROM ' . POSTS_TABLE . '
+					WHERE post_id = ' . (int) $attachment['post_msg_id'];
+				$result = $this->db->sql_query($sql);
+				$post_row = $this->db->sql_fetchrow($result);
+				$this->db->sql_freeresult($result);
+
+				if (!$post_row || !$this->content_visibility->is_visible('post', $post_row['forum_id'], $post_row))
+				{
+					// Attachment of a soft deleted post and the user is not allowed to see the post
+					throw new http_exception(404, 'ERROR_NO_ATTACHMENT');
+				}
+			}
+			else
+			{
+				// Attachment is in a private message.
+				$post_row = array('forum_id' => false);
+				$this->phpbb_download_handle_pm_auth( $attachment['post_msg_id']);
+			}
+
+			$extensions = array();
+			if (!extension_allowed($post_row['forum_id'], $attachment['extension'], $extensions))
+			{
+				throw new http_exception(403, 'EXTENSION_DISABLED_AFTER_POSTING', [$attachment['extension']]);
+			}
+		}
+
+		$display_cat = $extensions[$attachment['extension']]['display_cat'];
```

Comment:
I would like to know if i can remove everything related with this, and increment download count always except when is a thumbnail or is orphan.